x-pack/plugins/observability_solution/observability_onboarding/server/routes/flow/route.ts

```diff
@@ -171,8 +182,240 @@ const getProgressRoute = createObservabilityOnboardingServerRoute({
+ * product_service custom /path/to/error.log
+ * checkout_service custom /path/to/access.log
+ * checkout_service custom /path/to/error.log
+ * ```
+ *
+ * Example curl:
+ *
+ * ```bash
+ * curl --request POST \
+ *  --url "http://localhost:5601/internal/observability_onboarding/flow/${ONBOARDING_ID}/integrations/install" \
+ *  --header "Authorization: ApiKey ${ENCODED_API_KEY}" \
+ *  --header "Content-Type: text/tab-separated-values" \
+ *  --data $'system\tregistry\nproduct_service\tcustom\t/path/to/access.log\ncheckout_service\tcustom\t/path/to/access.log'
+ * ```
+ */
+const integrationsInstallRoute = createObservabilityOnboardingServerRoute({
+  endpoint: 'POST /internal/observability_onboarding/flow/{onboardingId}/integrations/install',
+  options: { tags: [], xsrfRequired: false },
+  params: t.type({
+    path: t.type({
+      onboardingId: t.string,
+    }),
+    body: t.string,
+  }),
+  async handler({ context, request, response, params, core, plugins, services }) {
+    const coreStart = await core.start();
+    const fleetStart = await plugins.fleet.start();
+    const savedObjectsClient = coreStart.savedObjects.createInternalRepository();
+    const packageClient = fleetStart.packageService.asScoped(request);
+
+    const savedObservabilityOnboardingState = await getObservabilityOnboardingFlow({
+      savedObjectsClient,
+      savedObjectId: params.path.onboardingId,
+    });
+    if (!savedObservabilityOnboardingState) {
+      throw Boom.notFound(`Onboarding session '${params.path.onboardingId}' not found.`);
+    }
+
+    const integrationsToInstall = parseIntegrationsTSV(params.body);
+    if (!integrationsToInstall.length) {
+      return response.badRequest({
+        body: {
+          message: 'Please specify a list of integrations to install',
+        },
+      });
+    }
+
+    await saveObservabilityOnboardingFlow({
+      savedObjectsClient,
+      savedObjectId: params.path.onboardingId,
+      observabilityOnboardingState: {
+        ...savedObservabilityOnboardingState,
+        type: 'logFiles',
+        progress: {},
+      } as ObservabilityOnboardingFlow,
+    });
+
+    let agentInputs: unknown[];
+    try {
+      agentInputs = await ensureInstalledIntegrations(integrationsToInstall, packageClient);
+    } catch (error) {
+      if (error instanceof FleetUnauthorizedError) {
+        return response.forbidden({
+          body: {
+            message: error.message,
+          },
+        });
+      }
+      throw error;
+    }
+
+    const authApiKey = getAuthenticationAPIKey(request);
+    const elasticsearchUrl = plugins.cloud?.setup?.elasticsearchUrl
+      ? [plugins.cloud?.setup?.elasticsearchUrl]
+      : await getFallbackESUrl(services.esLegacyConfigService);
+
+    return response.ok({
+      headers: {
+        'content-type': 'application/yaml',
+      },
+      body: generateAgentConfig({
+        apiKey: authApiKey ? `${authApiKey?.apiKeyId}:${authApiKey?.apiKey}` : '$API_KEY',
```

Comment:
@thomheymann is this the right api key? the one we are using for the installation of the integrations is not the one we should use to ship the data as it has too many privileges.

Reply:
No, this is currently not the correct key yet. We first need to confirm the approach and separate out the fleet privileges (#183020)